### ISSUE TITLE
git archive of a subdirectory of the repo does not include LFS files (you get pointers instead)

### ISSUE THREAD

**Issue by gknispel-australdx** [git-lfs/git-lfs#5343]
**Describe the bug**

If I archive with e.g. `git archive --prefix=repro/ -o ../repro.tgz HEAD` or even `git archive --prefix=repro/ -o ../repro.tgz HEAD  src/  ` the lfs files are properly included in the archive.

However, if I archive with e.g. `git archive --prefix=repro/ -o ../repro.tgz HEAD:src`  (which does not have the same effect as `git archive --prefix=repro/ -o ../repro.tgz HEAD  src/  `; the `HEAD:src` param will strip the `src/` prefix in all paths of the archive, while `HEAD src/` will select `src/` but not strip the `src/` prefix from the paths)

**To Reproduce**
1. Clone a repository which contains LFS files.
2. Create an archive by selecting a sub root for what to put in it with a syntax like `HEAD:src`
3. Unpack the archive elsewhere
4. The LFS files are replaces by their pointers

**Expected behavior**
I expected to get the actual content for the archive LFS files, and not their pointers.

**System environment**
Tested on Ubuntu 20.04 and Debian 11.

**Output of `git lfs env`**
```
git-lfs/2.13.2 (GitHub; linux amd64; go 1.15.9)
git version 2.30.2

Endpoint=https://gitlab.com/australdx/private/repro.git/info/lfs (auth=none)
  SSH=git@gitlab.com:australdx/private/repro.git
LocalWorkingDir=/home/user/git/repro
LocalGitDir=/home/user/git/repro/.git
LocalGitStorageDir=/home/user/git/repro/.git
LocalMediaDir=/home/user/git/repro/.git/lfs/objects
LocalReferenceDirs=
TempDir=/home/user/git/repro/.git/lfs/tmp
ConcurrentTransfers=8
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=/home/user/git/repro/.git/lfs
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic,lfs-standalone-file
UploadTransfers=basic,lfs-standalone-file
GIT_EXEC_PATH=/usr/lib/git-core
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"
```

**Comment by bk2204** [comment]
Hey,

This isn't a bug, but rather a limitation of how Git and Git LFS interact.  Git knows which files are LFS files by the `.gitattributes` file, which is typically in the root of the repository.  When you specify `HEAD:src`, Git resolves that to the given tree and archives it without consulting anything else in the commit (in other words, it doesn't consider `HEAD:.gitattributes`).  As a result, Git doesn't see the `.gitattributes` file in the root of the repository and doesn't invoke Git LFS for the pointer files.

In this case, Git LFS never actually gets invoked, so there's nothing that we can do to change how this works.  You can change this by putting a `.gitattributes` file in each subdirectory you want to archive, but that's the only way that this will work with Git.

**Comment by gknispel-australdx** [comment]
Hi, thanks for your explanation. This is unfortunate that it can't be made to work, but I understand the context. Maybe a warning about this limitation in the doc would be useful.

For reference, I managed to workaround for what I wanted to do by using a command like `git archive -o ../repro.tgz HEAD src/` and then post-processing the resulting tar to rename prefix `src/` to `repro/` using a small Python program (I did not want to unpack and repack manually to avoid messing up with potential metadata in the archive) doing essentially:

```
tar = tarfile.open(filename, mode='r')
output = tarfile.open(mode='w|', fileobj=sys.stdout.buffer)

for member in tar.getmembers():
    if member.name.startswith(path):
        if replacement:
            member.name = replacement + member.name[len(path):]
        output.addfile(member, tar.extractfile(member))

tar.close()
output.close()
```
